src/Pact/Typechecker.hs

```diff
@@ -266,8 +267,7 @@ applySchemas Pre ast = case ast of
     return ast
   (Binding _ bs _ (BindSchema n)) -> findSchema n $ \sch -> do
     debug $ "applySchemas: " ++ show (n,sch)
-    pmap <- M.fromList <$> forM bs
-      (\(Named _ node ni, Prim _ (PrimLit (LString bn))) -> (bn,) <$> ((Var node,ni,) <$> lookupTy node))
+    pmap <- M.fromList <$> traverseMaybe f bs
     validateSchema sch pmap
```

Comment:
Huh? What is the change here?

Reply:
The use of `traverseMaybe` and the `f` function below to avoid a non-exhaustive pattern match warning. Surely this is preferable to potential runtime failures.

Reply:
OH you defined `f` below, I was confused. Can you give it a name like matchString or something?

Reply:
Absolutely - using `f` was me being spartan out of habit.